scripts/plugin_runner/python_version/determine_python_version.py

```diff
@@ -0,0 +1,215 @@
+"""
+This script tries to determine the Python version for projects from a dataset using metadata from PyPI.
+
+It accepts
+    * path to the folder with python projects.
+    * path to `csv` file, where to save versions.
+
+First, we try to determine the version using the classifiers from the `setup.py` or `setup.cfg` files. If this fails,
+for each project we collect the requirements and use their classifiers specified on the PyPI to determine the version.
+"""
+
+import argparse
+import logging
+import re
+from pathlib import Path
+from typing import List, Optional, Set
+
+import pandas as pd
+
+from plugin_runner.python_version.python_classifiers import PythonClassifiers, PythonVersion
+
+from utils.file_utils import FileSystemItem, get_all_file_system_items
+from utils.python.pypi_utils import get_available_versions, get_package_classifiers
+from utils.python.requirements_utils import Requirements, gather_requirements
+
+logger = logging.getLogger(__name__)
+
+
+def _try_to_find_version_in_setup_file(file_path: Path) -> Optional[PythonVersion]:
+    """
+    Try to find the Python version using the PyPI classifiers in the setup file.
+
+    :param file_path: The path to the setup file.
+    :return: The found version of Python. If no Python version could be found, None will be returned.
+    """
+    with open(file_path) as file:
+        content = file.read()
+
+    if PythonClassifiers.PYTHON_3_ONLY.value in content:
+        return PythonVersion.PYTHON_3
+
+    if PythonClassifiers.PYTHON_2_ONLY.value in content:
+        return PythonVersion.PYTHON_2
+
+    for python_version in PythonVersion:
+        if any(classifier in content for classifier in PythonClassifiers.get_classifiers_by_version(python_version)):
+            return python_version
+
+    return None
+
+
+def _try_to_determine_version_using_classifiers(classifiers: Set[str]) -> Optional[PythonVersion]:
+    """
+    Try to determine the Python version using the classifiers obtained from the PyPI package metadata.
+
+    :param classifiers: List of PyPI classifiers.
+    :return: Python version. If it could not be determined will be returned None.
+    """
+    is_python_3 = any(
+        version_classifier in PythonClassifiers.get_classifiers_by_version(PythonVersion.PYTHON_3)
+        for version_classifier in classifiers
+    )
+
+    is_python_2 = any(
+        version_classifier in PythonClassifiers.get_classifiers_by_version(PythonVersion.PYTHON_2)
+        for version_classifier in classifiers
+    )
```

Comment:
It seems you can create a common functions and use here with python version as an argument

Reply:
I fixed it in a slightly different way: now we retrieve from the classifiers all the versions of python that are there, and if there is exactly one version, we return it, otherwise we return None.